pkg/reconciler/pipelinerun/resources/pipelinerunresolution.go

```diff
@@ -177,31 +252,36 @@ func (state PipelineRunState) IsBeforeFirstTaskRun() bool {
 }
 
 // IsStopping returns true if the PipelineRun won't be scheduling any new Task because
-// at least one task already failed or was cancelled in the specified dag
-func (state PipelineRunState) IsStopping(d *dag.Graph) bool {
-	for _, t := range state {
-		if _, ok := d.Nodes[t.PipelineTask.Name]; ok {
-			if t.IsCancelled() {
-				return true
-			}
-			if t.IsFailure() {
-				return true
-			}
-		}
-	}
-	return false
+// at least one task already failed or was cancelled in the specified dag, but there are
+// still tasks to be run in the DAG
+func (rpr ResolvedPipelineRun) IsStopping() bool {
+	return rpr.DagState.HasFailedOrCancelled() && !rpr.isDAGDone()
```

Comment:
There is a key change here - `IsStopping` will return true only as long at something is running in the main DAG.
As soon as the DAG is complete it will turn false, and will stay false while finally task are running.